src/dataframe.js

```diff
@@ -728,50 +728,21 @@ class DataFrame {
         return this.__newInstance__(this[__rows__], newColumnNames);
     }
 
-    /**
-     * Rename each column.
-     * @param {Array} newColumnNames The new column names of the DataFrame.
-     * @returns {DataFrame} A new DataFrame with the new column names.
-     * @example
-     * df.renameAll(['column1', 'column3', 'column4'])
-     */
-    renameAll(newColumnNames) {
-        if (newColumnNames.length !== this[__columns__].length) {
-            throw new WrongSchemaError(newColumnNames, this[__columns__]);
-        }
-        return this.__newInstance__(this.toArray(), newColumnNames);
-    }
-
-    /**
-     * Rename a column.
-     * @param {String} columnName The column to rename.
-     * @param {String} replacement The new name for the column.
-     * @returns {DataFrame} A new DataFrame with the new column name.
-     * @example
-     * df.rename('column1', 'columnRenamed')
-     */
-    rename(columnName, replacement) {
```

Comment:
The current usage of rename must be kept.
Indeed we deprecate this usage (we don't delete it completely).
The renameAll must not be deleted but deprecated.
It means that the method must remain compatible with the current usage.The current usage will be completely deleted on the next major version (2.0.0).
Moreover the method must throw a warning when used with the current usage.

As a solution, you can check if columnsMap is an object or a string.
If it's a string you could kept the current usage.
Else use your new usage.

Reply:
(However you can update the docstring with your new usage)

Reply:
Hi @Gmousse  Sorry I'm a lil confused. 


> As a solution, you can check if columnsMap is an object or a string.
> If it's a string you could kept the current usage.
> Else use your new usage
> 

Didn't understand the above. Is it for rename or renameAll? cuz rename takes 2 params currently.

Sorry for the back and forth questions

Reply:
Hi ! 
Sorry to be late !

I was talking about .rename. Indeed we will handle parameters in order to ensure compatibility.

Example of implementation:

```js
rename(mapping, deprecatedParam) {
    if (typeof mapping === "string") {
         // old behaviour
    } else {
        // your behaviour
    }
}
```